### ISSUE TITLE
[1.2] NoJavaUtilDate false-positive

### ISSUE THREAD

**Issue by boris-petrov** [CodeNarc/CodeNarc#349]
```groovy
import java.time.LocalDate
import com.company.my_custom.Date

Date makeDate(LocalDate date) {
	new Date(date)
}
```

This gives wrongly a `NoJavaUtilDate` warning.

**Comment by chrismair** [comment]
We could probably address a subset of these incorrect violations by scanning the imports. But I am a bit hesitant about doing that since that would not avoid violations if the `Date` class was defined within the same package. And this problem seems like a relatively uncommon edge case -- maybe better to just disable the rule (either per class or globally)?

**Comment by boris-petrov** [comment]
Scanning the imports sounds like a good idea to me.

The `Date` class being defined in the same packages is almost the same as the [other issue I opened](https://github.com/CodeNarc/CodeNarc/issues/353) and perhaps can be solved in a similar way?

Why would you think this is an uncommon edge case? Whenever someone has a custom `Date` class, this false-positive will be triggered. Scanning the imports will perhaps solve the issue completely. I don't see any negative impacts this could have. :)

**Comment by chrismair** [comment]
Okay, you convinced me. I have fixed it so that it ignores `new Date()` if the class imports another `Date` class.

**Comment by chrismair** [comment]
Available in CodeNarc 1.2.1
